src/Symfony/Component/Console/Command/InvokableCommand.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Command;
+
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Attribute\Argument;
+use Symfony\Component\Console\Attribute\Option;
+use Symfony\Component\Console\Exception\RuntimeException;
+use Symfony\Component\Console\Input\InputDefinition;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+/**
+ * Represents an invokable command.
+ *
+ * @author Yonel Ceruto <open@yceruto.dev>
+ *
+ * @internal
+ */
+class InvokableCommand
+{
+    public function __construct(
+        private readonly Command $command,
+        private readonly \Closure $code,
+    ) {
+    }
+
+    /**
+     * Invokes a callable with parameters generated from the input interface.
+     */
+    public function __invoke(InputInterface $input, OutputInterface $output): int
+    {
+        $statusCode = ($this->code)(...$this->getParameters($input, $output));
+
+        return is_numeric($statusCode) ? (int) $statusCode : 0;
```

Comment:
What about throwing an exception instead if it's not an integer?

Reply:
Makes sense to me. Updated!

Reply:
It seems to me that this is a breaking change for closures passed with `$command->setCode()`. The `InvokableCommand` class can take a constructor option to configure this behavior.

Reply:
Indeed, we missed this case when deprecating not returning an int from regular commands. I'd go with a deprecation instead of an exception then.

Reply:
We need an exception for the new invokable commands, and a deprecation for the closure code. There also an issue with closure not using parameter types.

This simple code is currently supported and will no longer:
```php
$command->setCode(function ($input, $output) {
   $output->print('Success!');
}
```

Do we really want to for projects to upgrade to this more verbose version:
```php
$command->setCode(function (InputInterface $input, OutputInterface $output): int {
   $output->print('Success!');

   return Command::SUCCESS;
}
```

Maybe, remove support of the new attributes for the closure code from this PR and create a follow-up PR to discuss this backward-compatibility specificities.

Reply:
For the return type it's allowed to return void (which fallback to `0`), no need to force return `int` in my opinion.

Reply:
And regular commands are already forced to return an `int` due the `execute(...): int` signature

Reply:
https://github.com/symfony/symfony/pull/59340#issuecomment-2578118660